### ISSUE TITLE
Product Collection (Beta) block if placed in mini-cart as cross sell doesn't allow add to cart

### ISSUE THREAD

**Issue by ResponsiveFlow** [woocommerce/woocommerce#46722]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I use the Product Collection (beta) block in the mini cart to show one additional product as a cross sell.<br>Unfortunately in the last few versions of WooCommerce, the add to cart button in the mini cart doesn't work anymore.

### Expected behavior

The add to cart button should work as expected and add the product to the cart.

### Actual behavior

The button is disabled

### Steps to reproduce

1. open the site editor
2. go to patterns > mini cart
3. in the mini cart add the Cross Sell block
4. go to the frontend, add a product to the cart and then open  the min cart.
5. The block isn't visible.

[https://github.com/woocommerce/woocommerce/assets/47855768/9db5a2f5-08fd-42fc-af17-b5a04c60e929](<https://github.com/woocommerce/woocommerce/assets/47855768/9db5a2f5-08fd-42fc-af17-b5a04c60e929>)

WordPress Environment

```
### WordPress Environment ###

WordPress address (URL): http://atelierdellaluce.com
Site address (URL): http://atelierdellaluce.com
WC Version: 8.8.2
REST API Version: ✔ 8.8.2
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.5.2
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.33
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 8.2.1
OpenSSL/3.1.1

SUHOSIN Installed: –
MySQL Version: 5.7.44-48-log
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.9.0
WC Database Prefix: bqf_
Total Database Size: 16.71MB
Database Data Size: 13.34MB
Database Index Size: 3.37MB
bqf_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_actions: Data: 0.13MB + Index: 0.14MB + Engine InnoDB
bqf_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_logs: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
bqf_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
bqf_jetpack_sync_queue: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_automations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_automation_runs: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_automation_run_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_automation_run_subjects: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_automation_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_automation_versions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletters: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_statistics_bounces: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_subscribers: Data: 0.02MB + Index: 0.23MB + Engine InnoDB
bqf_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_subscriber_tag: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_tags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_user_agents: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_options: Data: 4.08MB + Index: 0.08MB + Engine InnoDB
bqf_postmeta: Data: 0.16MB + Index: 0.16MB + Engine InnoDB
bqf_posts: Data: 4.50MB + Index: 0.06MB + Engine InnoDB
bqf_sgs_log_events: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
bqf_sgs_log_visitors: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
bqf_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
bqf_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
bqf_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
bqf_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 32
mailpoet_page: 1
page: 9
post: 3
product: 56
product_variation: 15
revision: 344
shop_order_placehold: 9
wp_font_face: 33
wp_font_family: 10
wp_global_styles: 2
wp_navigation: 2
wp_template: 6
wp_template_part: 6

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (7) ###

Gutenberg: by Gutenberg Team – 18.1.2
Jetpack: by Automattic – 13.3.1
Speed Optimizer: by SiteGround – 7.5.0
Security Optimizer: by SiteGround – 1.4.13
Ajax add to cart for WooCommerce: by QuadLayers – 2.3.4
WooPayments: by Automattic – 7.5.0
WooCommerce: by Automattic – 8.8.2

### Inactive Plugins (6) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.2
Code Snippets: by Code Snippets Pro – 3.6.4
MailPoet: by MailPoet – 4.48.2
WooCommerce Beta Tester: by WooCommerce – 2.3.0
WooCommerce Legacy REST API: by WooCommerce – 1.0.1
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 560 KB

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: ❌ Page does not contain the [woocommerce_checkout] shortcode or the woocommerce/checkout block.
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Three
Version: 1.4
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: /home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/error.php
/home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/success.php


### WooPayments ###

Version: 7.5.0
Connected to WPCOM: Yes
WPCOM Blog ID: 230817479
Account ID: ❌ Not connected
Logging: Disabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
product-pre-publish-modal
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-custom-fields
settings
async-product-editor-category-field
launch-your-store

Daily Cron: ✔ Next scheduled: 2024-04-18 13:40:23 +00:00
Options: ✔
Notes: 73
Onboarding: completed

### Action Scheduler ###

Complete: 287
Oldest: 2024-03-18 22:45:27 +0000
Newest: 2024-04-18 12:53:48 +0000

Failed: 1
Oldest: 2024-03-29 10:43:36 +0000
Newest: 2024-03-29 10:43:36 +0000

Pending: 2
Oldest: 2024-04-19 00:53:48 +0000
Newest: 2024-04-19 05:50:10 +0000


### Status report information ###

Generated at: 2024-04-18 13:25:12 +00:00
```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/products/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by Aljullu** [comment]
Hi there, @tomxygen! Thanks for opening this issue, we were able to reproduce it.

This is a tricky one, as the Mini-Cart block was built when the Interactivity API didn't exist, so they aren't compatible right now.

I will post here what we have found so far:

**Problem 1**

The Mini-Cart template part contents are being [filtered by `wp_kses_post()`](https://github.com/woocommerce/woocommerce/blob/d21b03588aa06ef733fc2ac72731ec771e370327/plugins/woocommerce/src/Blocks/BlockTypes/MiniCart.php#L495), which removes some HTML attributes used by the Interactivity API, the ones with double dashes `--`.

**Problem 2**

The Mini-Cart block [copies the HTML from the template part inside the drawer](https://github.com/woocommerce/woocommerce/blob/5bc0a2c8b1c4b0e10a2595dbeadbe7f16eedc776/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/component-frontend.tsx#L62-L64) and that's causing the Interactivity API not to work, I guess because JS event handlers aren't copied alongside the HTML.

**Solutions**

Problem 1 could be solved removing `wp_kses_post()`, as I don't think it's necessary in that context. However, in future versions of WordPress it's possible that `wp_kses_post()` will allow attributes with double dashes.

Problem 2 is a bit more complex and will likely require refactoring the Mini-Cart block so instead of copying the HTML contents, it attaches them to the drawer, something like `parent.appendChild( child )`. However, that's a bit tricky because it can bring issues if there are two Mini-Cart blocks rendered on the same screen.

@ralucaStan I reassigned this issue to Rubik, as I think both solutions would need to be applied in the Mini-Cart block code. In fact, the issue is not related to the Product Collection block itself: the Mini-Cart block is incompatible with any inner block using the Interactivity API. But please, let me know if you disagree.

**Comment by ralucaStan** [comment]
Thank you for sharing your findings @Aljullu.

This issue surfaced that a lot of blocks can be inserted in the Mini Cart drawer; see #46726

I agree that the fix should be part of the Mini Cart; considering the incompatibility of Product Collection (Beta) and the Mini Cart I think for now we should prevent it from being used in the Mini Cart until we can figure out the next steps.

Considering our team's limited knowledge with the Interactivity API a solution is not straight forward for now and we should spend some time in the next cycle trying to understand the problem better. @pmcpinto  for awarness

**Comment by ResponsiveFlow** [comment]
Thank you for the update @ralucaStan 
When is the next cycle going to happen?

**Comment by ralucaStan** [comment]
That would be beginning of June

**Comment by luisherranz** [comment]
> However, in future versions of WordPress it's possible that wp_kses_post() will allow attributes with double dashes

As far as I know, this was fixed in WP 6.6:

- https://github.com/WordPress/wordpress-develop/pull/6598
- https://core.trac.wordpress.org/ticket/61052

**Comment by ralucaStan** [comment]
I marked this ticket as blocked by https://github.com/woocommerce/woocommerce/issues/56450

**Comment by pierorocca** [comment]
Closing as this is no longer reproducible.

**Comment by pierorocca** [comment]
Thanks for flagging that @ralucaStan. There's a special tool we've been given for this triaging blitz and it lacks Linear links/comments which clearly is a problem.

**Comment by gigitux** [comment]
Thanks for opening this issue. I'm not able to reproduce this issue with WooCommerce 11.1:

https://github.com/user-attachments/assets/a00a85ba-f63a-42fa-a01e-f3ebb7858944

**Comment by mrwweb** [comment]
@gigitux What version of Woo are you running and is that product collection showing cross sells based on the cart contents? I'm still not able to get this working.

**Comment by gigitux** [comment]
> [@gigitux](https://github.com/gigitux) What version of Woo are you running and is that product collection showing cross sells based on the cart contents? I'm still not able to get this working.

Thanks for the additional context! I followed the reproduction steps that suggest adding a "Product Collection" block, so I picked the featured Products.

I'm able to reproduce this issue with the Cross Sells block. I'm going to reopen this issue and update the reproduction steps. Thanks for the feedback, @mrwweb!

**Comment by mrwweb** [comment]
Thanks @gigitux. I think it's plausible that there were overlapping issues and that the primary issue on this ticket was indeed solved (I genuinely don't know). It may make more sense to reopen #61081 which was closed as a duplicate of this issue.

All I care about is making sure the cross-sells issue gets fixed. [As I noted](https://github.com/woocommerce/woocommerce/issues/61081#issuecomment-3657693976) on that issue, it appears that the Woo-provided setting for cart-based cross-sells simply doesn't work.
